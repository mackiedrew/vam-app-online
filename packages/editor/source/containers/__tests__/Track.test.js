import {
  Track as Subject,
  mapStateToProps,
  mapDispatchToProps
} from "../Track";
jest.mock("../TrackControls", () => "TrackControls");
jest.mock("../../components/Waveform", () => "Waveform");
jest.mock("../../help/wav", () => ({
  richReadWav: () =>
    new Promise(resolve =>
      resolve({
        sampleRate: 0,
        grains: []
      })
    )
}));

jest.mock("../../selectors/getGrainTagsFactory", () => () => () => []);

const mockSettings = {
  grain: {
    label: "Grain Time",
    value: 5,
    unit: "s",
    type: "number"
  },
  quietCutoff: {
    label: "Quietness Threshold",
    value: 10,
    unit: "%",
    type: "number"
  }
};

const mockTracks = {
  "123ABC": {
    grains: [{ start: 0, end: 4 }, { start: 4, end: 8 }],
    fileName: "test.wav"
  }
};

const mockProps = {
  id: "123ABC",
  seekPosition: 10,
  selectedTrack: "123ABC",
  settings: mockSettings,
  trackList: mockTracks,
  view: { start: 0, end: 20 },
  maxAmplitudes: { "123ABC": 0 },
  grainTags: [{ quiet: true }, { quiet: false }]
};

describe("<Track />", () => {
  it("renders without crashing", () => {
    shallow(<Subject {...mockProps} />);
  });

  describe("renders correctly", () => {
    it("with no grains", () => {
      const tree = renderer
        .create(<Subject {...mockProps} trackList={{ "123ABC": {} }} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("with grains", () => {
      const tree = renderer.create(<Subject {...mockProps} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it("readPath() calls all actions it needs to", () => {
    const mockSetTrackSampleRate = sinon.spy();
    const mockSetTrackGrains = sinon.spy();
    const subject = shallow(
      <Subject
        {...mockProps}
        setTrackGrains={mockSetTrackGrains}
        setTrackSampleRate={mockSetTrackSampleRate}
      />
    );
    return subject.instance().readPath().then(() => {
      expect(mockSetTrackSampleRate.called).toBe(true);
      expect(mockSetTrackGrains.called).toBe(true);
    });
  });

  describe("mapStateToProps()", () => {
    const mockState = {
      tracks: {
        seekPosition: 1,
        trackList: 2,
        selectedTrack: 3,
        view: 4
      },
      settings: 5
    };

    it("returns object with expected properties", () => {
      const {
        seekPosition,
        selectedTrack,
        settings,
        trackList,
        view
      } = mapStateToProps(mockState, { id: "123ABC" });
      expect(seekPosition).toBe(1);
      expect(trackList).toBe(2);
      expect(selectedTrack).toBe(3);
      expect(view).toBe(4);
      expect(settings).toBe(5);
    });
  });

  describe("mapDispatchToProps()", () => {
    it("returns object with expected properties", () => {
      const mockDispatch = sinon.spy();
      const {
        setTrackSampleRate,
        setTrackGrains,
        setSeekPosition
      } = mapDispatchToProps(mockDispatch);
      expect(typeof setTrackSampleRate).toBe("function");
      expect(typeof setTrackGrains).toBe("function");
      expect(typeof setSeekPosition).toBe("function");
    });
  });
});
