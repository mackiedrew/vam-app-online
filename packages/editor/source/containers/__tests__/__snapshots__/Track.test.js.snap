// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> renders correctly with grains 1`] = `
<div
  className="track"
  id="track-123ABC"
>
  <TrackControls
    id="123ABC"
    name="test.wav"
  />
  <div
    className="display"
  >
    
    <div
      className="seek-line"
      style={
        Object {
          "left": "50%",
        }
      }
    />
    <Waveform
      grainTags={
        Array [
          Object {
            "quiet": true,
          },
          Object {
            "quiet": false,
          },
        ]
      }
      grains={undefined}
      maxAmplitude={0}
      selected={true}
      setSeekPosition={undefined}
      view={
        Object {
          "end": 20,
          "start": 0,
        }
      }
    />
  </div>
</div>
`;

exports[`<Track /> renders correctly with no grains 1`] = `
<div
  className="track"
  id="track-123ABC"
>
  <TrackControls
    id="123ABC"
    name={undefined}
  />
  <div
    className="display"
  >
    <div
      className="loading"
      id="loading"
    >
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
      <div
        className="bar"
      />
    </div>
    <div
      className="seek-line"
      style={
        Object {
          "left": "50%",
        }
      }
    />
    <Waveform
      grainTags={
        Array [
          Object {
            "quiet": true,
          },
          Object {
            "quiet": false,
          },
        ]
      }
      grains={undefined}
      maxAmplitude={0}
      selected={true}
      setSeekPosition={undefined}
      view={
        Object {
          "end": 20,
          "start": 0,
        }
      }
    />
  </div>
</div>
`;
